src/main/java/com/simibubi/create/content/kinetics/chainConveyor/ChainConveyorRenderer.java

```diff
@@ -131,8 +155,88 @@ private void renderBox(ChainConveyorBlockEntity be, PoseStack ms, MultiBufferSou
 		}
 	}
 
+	/**
+	 * Calculate the intersection points between a line segment and a circle centered at cameraPos with radius LODDistance.
```

Comment:
circle or sphere?

but why do we need to do all this complex math at all? can't we just cutover to the LOD chain after a specific depth? then there's only ever 2 segments of chain and we don't have to worry about all the different cases of intersection counts